R/Figures.R

```diff
@@ -20,24 +22,33 @@
 #' # down to one Comparable Data Identifier
 #' 
 #' # Create boxplot:
-#' TADA_boxplot(TADAProfileClean18_TNonly)
+#' TADA_boxplot(TADAProfileClean18_TNonly, id_col = "TADA.ComparableDataIdentifier")
 #' 
 
-TADA_boxplot <- function(filtered.data) {
+TADA_boxplot <- function(filtered.data, id_col = c("TADA.CharacteristicName", "TADA.ComparableDataIdentifier")) {
   # check .data is data.frame
   checkType(filtered.data, "data.frame", "Input object")
+  # check id_col matches one of the options
+  id_col <- match.arg(id_col)
+  # check .data has required columns
+  checkColumns(filtered.data, id_col)
   # check .data has required columns
-  checkColumns(filtered.data, c("TADA.ComparableDataIdentifier", "TADA.ResultMeasureValue", "TADA.ResultMeasure.MeasureUnitCode"))
-  # check TADA.ComparableDataIdentifier column is filtered to one identifier
-  if (length(unique(filtered.data$TADA.ComparableDataIdentifier)) > 1) {
-    warning("Boxplot function cannot run with more than 1 comparable data identifier. Please filter dataframe and rerun function.")
+  checkColumns(filtered.data, c("TADA.ResultMeasureValue", "TADA.ResultMeasure.MeasureUnitCode"))
+  # check id_col is filtered to one characteristic or identifier
+  if (length(unique(filtered.data[,id_col])) > 1) {
+    stop(paste0(id_col, " field contains more than one unique value. Boxplot function cannot run with more than 1 unique characteristic or comparable data identifier. Please filter dataframe and rerun function."))
+  }
+  # check that units are the same across all data points
```

Comment:
Good check here. Adding note to discuss where the unit harmonization happens in the shiny app